### PR TITLE
Metaview

### PR DESCRIPTION
**Pull request by jhaux**
# Meta Views! Render a view of your data

Allows to specify a base dataset and a possibly nested set of views on
that dataset. The labels are rendered as specified in the view and
example loading is done per view on the base dataset.

Allowing nested labels makes it necessary that labels are always numpy arrays!
This is why a lot of tests are updated and some Dataset classes need
fixing.

Still missing:
- MetaViewDataset tests
- Good documentation with examples

### CODE REVIEW

**Comment by jhaux** [comment]
Before merging, let me add the option to pass config parameters for the base dataset via the `meta.yaml`.

File: edflow/data/dataset_mixin.py [hunk 1]
```diff
@@ -32,6 +33,11 @@ class DatasetMixin(DatasetMixin_):
     manipulate the dataset. When getting the actual example we can do the heavy
     lifting like loading and/or manipulating images.
 
+    .. warning:: 
+        Labels must be ``dict`` s of ``numpy`` arrays and not ``list`` s!
+        Otherwise many operations do not work and result in incoprehensible
```
Comment (pesser) [inline edflow/data/dataset_mixin.py]: incoprehensible -> incomprehensible

File: edflow/data/dataset_mixin.py [hunk 2]
```diff
@@ -32,6 +33,11 @@ class DatasetMixin(DatasetMixin_):
...
+        Errors.
```
Comment (pesser) [inline edflow/data/dataset_mixin.py]: Errors -> errors

File: edflow/data/dataset_mixin.py [hunk 3]
```diff
@@ -92,6 +98,12 @@ def __init__(self):
 
     **Labels in the example ``dict``**
 
+    Oftentimes it is good to store and load some values as lables as it can
+    increase performance and decrease storage size, e.g. when storing scalar
+    D = ConcatenatedDataset(A, A, A)  # Multiplying two datasets
```
Comment (pesser) [inline edflow/data/dataset_mixin.py]: did something go wrong here? when storing scalar D = ConcatenatedDataset ? Also the text is duplicated below in line 107?

**Review by pesser — APPROVED** [review]
nice new dataset! we really need an introduction to MetaDataset and MetaView to make sure people know about them.
Take a look at my comments regarding the doc and merge if you think it is ready!

File: edflow/data/dataset_mixin.py (hunk 1 above)
Comment (jhaux) [reply edflow/data/dataset_mixin.py]: Done!

File: edflow/data/dataset_mixin.py (hunk 2 above)
Comment (jhaux) [reply edflow/data/dataset_mixin.py]: Done!

File: edflow/data/dataset_mixin.py (hunk 3 above)
Comment (jhaux) [reply edflow/data/dataset_mixin.py]: Yeah, that was weird. It's fixed now.

**Pull request merged**